### ISSUE TITLE
Modifier keys are not released on Wayland

### ISSUE THREAD

**Issue by davuses** [k0kubun/xremap#139]
Hi, I'm running xremap under wayland, for me it doesn't work in gnome gtk apps. There's no issue with qt based apps or others.

I'm using Nvidia graphic card, not sure if it's something to do with that.

I'm on Ubuntu22.04, gnome 42.2, Wayland, NVIDIA Driver Version: 510.73.05

The configuration file:
```
keymap:
  - name: Arrow-remap
    remap:
      SHIFT-ALT-KEY_H: KEY_LEFT
      SHIFT-ALT-KEY_L: KEY_RIGHT
      SHIFT-ALT-KEY_K: KEY_UP
      SHIFT-ALT-KEY_J: KEY_DOWN
```

**Comment by k0kubun** [comment]
> gnome gtk apps

Can you report an actual app you're using? I have almost the same environment as yours, so I'd be able to test the same thing if I know what application should be tested.

**Comment by davuses** [comment]
> Can you report an actual app you're using? I have almost the same environment as yours, so I'd be able to test the same thing if I know what application should be tested.

It's not working in all gnome core apps, like nautilus file explorer, gedit text editor, or system setting.

**Comment by k0kubun** [comment]
Thanks for the information. I successfully reproduced the issue on Nautilus and system settings. xremap seems to generate intended events but somehow only those manipulated events are not recognized by those applications. I'll look into it, hopefully this weekend.

**Comment by davuses** [comment]
> 

Thanks!

**Comment by k0kubun** [comment]
I took a look at this today. What's happening seems to be that, when xremap generates a release event of a modifier key through evdev input, it's not recognized as a modifier release by Wayland unless it's under XWayland. As a workaround, I tried generating a synchronization event or sleeping for some milliseconds, but none of them worked well.

It seems like a bug in Wayland or at least something that Wayland handles differently from X11. I'll find a way to talk to the maintainers of Wayland and/or uinput to ask how this could be fixed.
